### PR TITLE
[FIX] Use client document version for edit

### PR DESCRIPTION
**Pull request by samuel-olivier**

### CODE REVIEW

File: src/language-client-mutualization.ts [hunk 1]
```diff
@@ -237,12 +237,16 @@ function bindClientToServer (
...
             : undefined,
-          documentChanges: params.edit.documentChanges?.filter(documentEdit => {
-            if (TextDocumentEdit.is(documentEdit)) {
-              return isDocumentOpen(documentEdit.textDocument.uri)
-            }
-            return false
-          })
+          documentChanges: params.edit.documentChanges
+            ?.filter(TextDocumentEdit.is)
+            .filter(({ textDocument }) => isDocumentOpen(textDocument.uri) && documents.get(textDocument.uri) != null)
```
Comment (CGNonofr) [inline src/language-client-mutualization.ts]: Why checking `documents.get(textDocument.uri) != null`? isn't already what isDocumentOpen is doing?

Comment (samuel-olivier) [reply src/language-client-mutualization.ts]: You're right, I didn't check !
[fix: remove unnecessary null check](https://github.com/CodinGame/languageserver-mutualized/pull/16/commits/5eda97f196731fa770256f739a00a0fe97b0d495)

**Comment by samuel-olivier** [comment]
@CGNonofr some comments: [docs: explain the choice about document version](https://github.com/CodinGame/languageserver-mutualized/pull/16/commits/6329b4eed7368b6d4ed6cd426cde899b6b6d59aa)

**Review by CGNonofr — COMMENTED** [review]
Thanks!

**Review by CGNonofr — APPROVED** [review]

**Pull request merged**